app/session.go

```diff
@@ -63,7 +63,7 @@ func (a *App) GetSession(token string) (*model.Session, *model.AppError) {
 		var err *model.AppError
 		session, err = a.createSessionForUserAccessToken(token)
 		if err != nil {
-			return nil, model.NewAppError("GetSession", "api.context.invalid_token.error", map[string]interface{}{"Token": token}, err.Error(), http.StatusUnauthorized)
+			return nil, model.NewAppError("GetSession", "api.context.invalid_token.error", map[string]interface{}{"Token": token}, "", http.StatusUnauthorized)
```

Comment:
My concern with a change like this is that it potentially "reads" as broken to a random observer. "Why isn't err added to the AppError? Surely that's a bug -- here, quick fix."

Could we perhaps instead test for the `app.user_access_token.invalid_or_missing` and return a replacement error of our own to make this explicit?

Reply:
Eeehhh I'm not sure I agree someone would interpret this as a bug, we commonly don't add the detailed error all over the codebase. We also do the exact same thing other places in this function. I do agree that checking the error ID for `app.user_access_token.invalid_or_missing` is probably a bit better from a code reader's perspective but we actually turn 401s into INFOs so I think it's better in this case to log the errors we care about separately and explicitly

Only really 1/5 so if you have a strong opinion I'll change it

Reply:
I guess I could check the error ID and log the error here instead of inside `createSessionForUserAccessToken`

Reply:
Fair enough re: not including the error in `AppError` -- I agree that was a stretch :)

I like your counter of checking the error ID and logging, as it makes this change explicit. 2/5?

Reply:
Ok cool, yeah I like that better too. I'll update it 👍